pilot/test/discovery/discovery_test.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package discovery_test
+
+import (
+	"testing"
+
+	"istio.io/istio/pilot/adapter/config/file"
+	"istio.io/istio/pilot/adapter/config/memory"
+	"istio.io/istio/pilot/model"
+	"istio.io/istio/pilot/test/discovery"
+)
+
+func TestMockPilot(t *testing.T) {
+	mockStore := memory.Make(model.IstioConfigTypes)
+	configStore := file.NewConfigStore(mockStore)
+
+	cfg := (&discovery.PilotConfig{}).
```

Comment:
And this is why golang doesn't use builder-like patterns :), it's just not ergonomic in the language.

Option API is best here, but another alternative that's still more idiomatic than this would be to provide methods with the defaults. Either a `NewMockPilot(configstore)` (probably preferred in this case), or:

```go
disc := discovery.MockDiscovery(), // ew, name stutter
cfg := &discobery.PilotConfig{
	ConfigCache:      configStore,
	ServiceDiscovery: disc,
	ServiceAccounts:  disc,
	Controller:       discovery.MockServiceController(),
	Mesh:             discovery.MockMesh(),
}
```